Add option to not reveal the instant or preliminary scores

peterdietz opened the pull request:
Add an admin-level option to choose to show or hide the preliminary scores. Also, updates the cache length time to be based upon preliminary score, vs 60 minute delay.

baholladay commented on db/install.xml:
```diff
@@ -9,13 +9,14 @@
-	<FIELD NAME="timeretrieved" TYPE="int" LENGTH="10" NOTNULL="false" UNSIGNED="true" DEFAULT="0" SEQUENCE="false" PREVIOUS="identifier" NEXT="data"/>
+        <FIELD NAME="identifier" TYPE="char" LENGTH="255" NOTNULL="true" SEQUENCE="false" PREVIOUS="userid" NEXT="similarityscore"/>
+        <FIELD NAME="similarityscore" TYPE="int" LENGTH="5" NOTNULL="true" UNSIGNED="true" DEFAULT="0" SEQUENCE="false" PREVIOUS="identifier" NEXT="preliminary"/>
+        <FIELD NAME="preliminary" TYPE="int" LENGTH="2" NOTNULL="true" UNSIGNED="true" DEFAULT="0" SEQUENCE="false" PREVIOUS="similarityscore" NEXT="timeretrieved"/>
```
Make sure to uninstall and drop the tables and reinstall VC to test this code

baholladay commented on lib.php:
```diff
@@ -206,24 +210,29 @@ public function get_file_results($cmid, $userid, $file, $vericite=null) {
-					//recheck after PLAGIARISM_VERICITE_SCORE_CACHE_IGNORE_MIN_RECHECK minutes instead of the normal cache time
-					$cacheTime = 60 * PLAGIARISM_VERICITE_SCORE_CACHE_IGNORE_MIN_RECHECK;
+
+				if(!isset($mycontent->preliminary) || $mycontent->preliminary==1){
```
I don't understand the "!isset" part of this condition

baholladay commented on lib.php:
```diff
@@ -206,24 +210,29 @@ public function get_file_results($cmid, $userid, $file, $vericite=null) {
 			//there should only be 1 result
             foreach ($contentscore as $content) {
                 $mycontent = $content;
-				if(isset($mycontent->timesubmitted) && (time() - $mycontent->timesubmitted) < (60 * PLAGIARISM_VERICITE_SCORE_CACHE_IGNORE_MIN)){
```
I noticed you removed the min recheck flag in these conditions. Have you watched the logs to make sure that we do not call our API too many times in cases with multiple submissions that are all outside of the cache limit? I believe that min recheck was a safe guard against that.

baholladay commented on lib.php:
```diff
@@ -206,24 +210,29 @@ public function get_file_results($cmid, $userid, $file, $vericite=null) {
-					$cacheTime = 60 * PLAGIARISM_VERICITE_SCORE_CACHE_IGNORE_MIN_RECHECK;
+
+				if(!isset($mycontent->preliminary) || $mycontent->preliminary==1){
+                    plagiarism_vericite_log("don't consult cache");
```
why do you log "don't consult cache, but you have a "cachetime" value > 0

baholladay commented on lib.php:
```diff
@@ -798,6 +812,24 @@ function plagiarism_vericite_ends_with($str, $test) {
+ */
+function plagiarism_vericite_is_cache_time_expired($timeRetrieved, $cacheTime) {
+    plagiarism_vericite_log("now(): " . time() . " retrieved: " . $timeRetrieved . " cacheTime: " . $cacheTime . " timeLeft: " . plagiarism_vericite_time_left_in_cache($timeRetrieved, $cacheTime));
+    if($timeRetrieved == 0) {
```
why is this needed? If timeRetrieved is 0, then time() - 0 should be greater than the cacheTime

baholladay commented:
Please try to undo all of the changes surrounding the cache timeout and make it as simple as possible like:

cacheTime = isPreliminary? PLAGIARISM_VERICITE_SCORE_CACHE_PRELIMINAYR_IGNORE_MIN : PLAGIARISM_VERICITE_SCORE_CACHE_IGNORE_MIN

baholladay commented on sdk/lib/Model/ReportScoreReponse.php:
```diff
@@ -69,7 +69,9 @@ class ReportScoreReponse implements ArrayAccess
-        'score' => 'int'
+        'preliminary' => 'bool',
+        'score' => 'int',
+        'user' => 'string'
```
user is here twice

baholladay commented on sdk/lib/Model/ReportScoreReponse.php:
```diff
@@ -85,7 +87,9 @@ public static function swaggerTypes()
+        'preliminary' => 'preliminary',
+        'score' => 'score',
+        'user' => 'user'
     );
```
user is here twice

baholladay commented on sdk/lib/Model/ReportScoreReponse.php:
```diff
@@ -101,7 +105,9 @@ public static function attributeMap()
+        'preliminary' => 'setPreliminary',
+        'score' => 'setScore',
+        'user' => 'setUser'
     );
```
user is in here twice

baholladay commented on sdk/lib/Model/ReportScoreReponse.php:
```diff
@@ -117,7 +123,9 @@ public static function setters()
+        'preliminary' => 'getPreliminary',
+        'score' => 'getScore',
+        'user' => 'getUser'
     );
```
user is in here twice

peterdietz committed:
Add option to not reveal preliminary scores

- Add plugin option to not show preliminary scores
- Add 'preliminary' field to PHP SDK
- Add 'preliminary' field to db cache plagiarism_vericite_files

peterdietz commented:
Pushed updated PR that is smaller.
